clippy_lints/src/redundant_clone.rs

```diff
@@ -766,3 +750,30 @@ impl PossibleBorrowerMap<'_, '_> {
         self.maybe_live.contains(local)
     }
 }
+
+#[derive(Default)]
+struct TransitiveRelation {
+    relations: FxHashMap<mir::Local, Vec<mir::Local>>,
+}
+impl TransitiveRelation {
+    fn add(&mut self, a: mir::Local, b: mir::Local) {
+        self.relations.entry(a).or_default().push(b);
+    }
+
+    fn reachable_from(&self, a: mir::Local, domain_size: usize) -> HybridBitSet<mir::Local> {
+        let mut seen = HybridBitSet::new_empty(domain_size);
+        seen.insert(a);
+        let mut stack = vec![a];
+        while let Some(u) = stack.pop() {
+            if let Some(edges) = self.relations.get(&u) {
+                for &v in edges {
+                    if seen.insert(v) {
+                        stack.push(v);
+                    }
+                }
+            }
+        }
+        seen.remove(a);
```

Comment:
Is it certain that `a` cannot be inserted on line 770?

Reply:
Thank you for the review! I was imagined that borrow-borrower relation is a DAG (directed acyclic graph), but I'm actually uncertain. I'll remove L:765 and L:776 so it would work even if it wasn't a DAG (with less than 2x traversal which is just a constant factor).